packages/forklift-console-plugin/webpack.config.ts

```diff
@@ -7,141 +7,183 @@ import CopyPlugin from 'copy-webpack-plugin';
-          fullySpecified: false,
         },
-      },
-    ],
-  },
-  devServer: {
-    static: ['./dist'],
-    host: 'localhost',
-    port: 9001,
-    headers: {
-      'Access-Control-Allow-Origin': '*',
-      'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, PATCH, OPTIONS',
-      'Access-Control-Allow-Headers': 'X-Requested-With, Content-Type, Authorization',
-      'Service-Worker-Allowed': '/', // needed to support MockServiceWorker
-    },
-    devMiddleware: {
-      writeToDisk: true,
+      ],
     },
-    client: {
-      progress: true,
+    devServer: {
+      static: ['./dist'],
+      host: 'localhost',
+      port: 9001,
+      headers: {
+        'Access-Control-Allow-Origin': '*',
+        'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, PATCH, OPTIONS',
+        'Access-Control-Allow-Headers': 'X-Requested-With, Content-Type, Authorization',
+        'Service-Worker-Allowed': '/', // needed to support MockServiceWorker
+      },
+      devMiddleware: {
+        writeToDisk: true,
+      },
+      client: {
+        progress: true,
+      },
     },
-  },
-  plugins: [
-    new DynamicConsoleRemotePlugin({
-      pluginMetadata,
-      extensions,
-    }),
-    new CopyPlugin({
-      patterns: [{ from: '../locales', to: '../dist/locales' }],
-    }),
-    new EnvironmentPlugin({
-      // DATA_SOURCE: used for testing when no api servers are available
-      // if set to mock, network api calls will be mocked
-      DATA_SOURCE: 'remote',
-      // can be 'RedHat' or 'Konveyor',
-      // note: downstream build are set to: 'RedHat'
-      BRAND_TYPE: 'Konveyor',
-      // NAMESPACE: used only on mock data
-      NAMESPACE: 'konveyor-forklift',
-      // DEFAULT_NAMESPACE: UI forms and modals will fallback to this namespace
-      //                    if no namespace is given by user.
-      // note: downstream build are set to: 'openshift-mtv'
-      DEFAULT_NAMESPACE: 'konveyor-forklift',
-      // NODE_ENV: used to bake debugging information on development builds.
-      NODE_ENV: production ? 'production' : 'development',
-      // PLUGIN_NAME: should be set to the plugin name hardcoded in the
-      //              installation scripts, defaults to 'forklift-console-plugin'.
-      PLUGIN_NAME: pluginMetadata.name,
-    }),
-  ],
-  devtool: 'source-map',
-  optimization: {
-    chunkIds: production ? 'deterministic' : 'named',
-    minimize: production ? true : false,
-    minimizer: [
-      // Keep class names and function names in sources to aid debug and diagnostics of prod builds
-      new TerserPlugin({
-        terserOptions: {
-          keep_classnames: true,
-          keep_fnames: true,
-        },
+    plugins: [
+      new DynamicConsoleRemotePlugin({
+        pluginMetadata,
+        extensions,
+      }),
+      new BannerPlugin({ banner, raw: true }),
+      new CopyPlugin({
+        patterns: [
+          { from: '../locales', to: '../dist/locales' },
+          {
```

Comment:
i prefer a build from scratch plugin, like we use in rollup, can you make a plugin similar to `writeJSONFile` that will add the build info without the need for a template ?

Reply:
I don't see the need to hide it, but I will look to see how involved it would be to do everything as another custom WebpackPlugin.

Reply:
Moved to a custom webpack plugin so now it is both less magical and more magical.

Reply:
Reworked yet again.